internal/output/json.go

```diff
@@ -59,7 +59,7 @@ func (f *JSON) Fprint(writer io.Writer, value interface{}) {
 	f.cfg.OutWriter.Write([]byte(nul + "\n")) // Terminate with NUL character so consumers can differentiate between multiple output messages
 }
 
-type jsonError struct {
+type JsonError struct {
```

Comment:
Now that we're making this a public struct I feel more opinionated about the naming; can we name this StructuredError instead? That'll keep the naming consistent. 

The fact is we currently only have JSON output so by definition structured errors are json errors, but that might not always be the case. I'd rather we future-proof the naming and thus clearly mark the boundaries between json and structured output.

Reply:
Okay, I've changed the name, but I left it in the output/json.go. I wasn't sure where exactly else it can/should go. Let me know if you have a preference.

I also removed the `[]string` possibility for `toStructuredError()` since the errors cmdlet returns a `StructuredError` directly.